packages/@react-aria/combobox/src/useComboBox.ts

```diff
@@ -119,6 +119,13 @@ export function useComboBox<T>(props: AriaComboBoxOptions<T>, state: ComboBoxSta
         state.commit();
         break;
       case 'Escape':
+        if (
+          state.selectedKey !== null ||
+          state.inputValue === '' ||
+          props.allowsCustomValue
```

Comment:
One weird case here is if you set a custom value, and then press escape, it both resets the combobox to its previous selected value, and also closes the dialog. Probably should only reset first, and then a second press should close it.

Also initially looking at the logic here it seemed like there should be some condition about the menu also being closed, but somehow that is already happening. Do you know why?